folding-schemes/src/folding/nova/mod.rs

```diff
@@ -249,6 +250,187 @@ where
+            .alpha
+            .serialize_with_mode(&mut writer, compress)?;
+        self.poseidon_config
+            .ark
+            .serialize_with_mode(&mut writer, compress)?;
+        self.poseidon_config
+            .mds
+            .serialize_with_mode(&mut writer, compress)?;
+        self.poseidon_config
+            .rate
+            .serialize_with_mode(&mut writer, compress)?;
+        self.poseidon_config
+            .capacity
+            .serialize_with_mode(&mut writer, compress)
+    }
+
+    fn serialized_size(&self, compress: ark_serialize::Compress) -> usize {
+        self.cs_params.serialized_size(compress)
+            + self.cf_cs_params.serialized_size(compress)
+            + self.i.serialized_size(compress)
+            + self.z_0.serialized_size(compress)
+            + self.z_i.serialized_size(compress)
+            + self.w_i.serialized_size(compress)
+            + self.u_i.serialized_size(compress)
+            + self.W_i.serialized_size(compress)
+            + self.U_i.serialized_size(compress)
+            + self.cf_W_i.serialized_size(compress)
+            + self.cf_U_i.serialized_size(compress)
+            + self.r1cs.serialized_size(compress)
+            + self.cf_r1cs.serialized_size(compress)
+            + self.poseidon_config.full_rounds.serialized_size(compress)
+            + self
+                .poseidon_config
+                .partial_rounds
+                .serialized_size(compress)
+            + self.poseidon_config.alpha.serialized_size(compress)
+            + self.poseidon_config.ark.serialized_size(compress)
+            + self.poseidon_config.mds.serialized_size(compress)
+            + self.poseidon_config.rate.serialized_size(compress)
+            + self.poseidon_config.capacity.serialized_size(compress)
+    }
+
+    fn serialize_compressed<W: Write>(
+        &self,
+        writer: W,
+    ) -> Result<(), ark_serialize::SerializationError> {
+        self.serialize_with_mode(writer, ark_serialize::Compress::Yes)
+    }
+
+    fn compressed_size(&self) -> usize {
+        self.serialized_size(ark_serialize::Compress::Yes)
+    }
+
+    fn serialize_uncompressed<W: Write>(
+        &self,
+        writer: W,
+    ) -> Result<(), ark_serialize::SerializationError> {
+        self.serialize_with_mode(writer, ark_serialize::Compress::No)
+    }
+
+    fn uncompressed_size(&self) -> usize {
+        self.serialized_size(ark_serialize::Compress::No)
+    }
+}
+
+impl<
+        C1: CurveGroup,
+        GC1: CurveVar<C1, CF2<C1>> + ToConstraintFieldGadget<CF2<C1>>,
+        C2: CurveGroup,
+        GC2: CurveVar<C2, CF2<C2>>,
+        FC: FCircuit<C1::ScalarField, Params = ()>,
+        CS1: CommitmentScheme<C1>,
+        CS2: CommitmentScheme<C2>,
+    > Nova<C1, GC1, C2, GC2, FC, CS1, CS2>
+where
+    CS1::ProverParams: CanonicalDeserialize,
+    CS2::ProverParams: CanonicalDeserialize,
+{
+    #[allow(dead_code)]
+    fn deserialize_nova<R: std::io::prelude::Read>(
```

Comment:
We can't derive or implement the `CanonicalDeserialize` trait since `CurveVar` notably does not implement the `Sync` trait, which is required.